src/Codeception/Configuration.php

```diff
@@ -159,39 +169,45 @@ public static function config(string $configFile = null): array
         if ($dir !== false) {
             self::$dir = $dir;
             $configDistFile = $dir . DIRECTORY_SEPARATOR . 'codeception.dist.yml';
+
+            // set the one default base directory for included setup
+            if (!self::$baseDir) {
+                self::$baseDir = $dir;
+            }
         }
 
         if (!file_exists($configFile) && (!isset($configDistFile) || !file_exists($configDistFile))) {
             throw new ConfigurationException("Configuration file could not be found.\nRun `bootstrap` to initialize Codeception.", 404);
         }
 
-        // set the one default base directory for included setup
-        if (!self::$baseDir) {
-            self::$baseDir = $dir;
-        }
-
         // Preload config to retrieve params such that they are applied to codeception config file below
         $tempConfig = self::$defaultConfig;
 
         $distConfigContents = '';
         if (isset($configDistFile) && file_exists($configDistFile)) {
             $distConfigContents = file_get_contents($configDistFile);
+            if ($distConfigContents === false) {
+                throw new ConfigurationException("Failed to read {$configDistFile}");
```

Comment:
Hmm... Is it possible for us to provide information about *why* it failed using the [`error_get_last`](https://www.php.net/manual/es/function.error-get-last.php) function?

Reply:
Error handler displays PHP Warning anyway.

I did `chmod 000 codeception.yml` and test it with Codeception before and after making this change.
Before:
```
PHP Warning:  file_get_contents(/.../module-webdriver/codeception.yml): Failed to open stream: Permission denied in /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Configuration.php on line 184
PHP Stack trace:
PHP   1. {main}() /.../module-webdriver/vendor/bin/codecept:0
PHP   2. eval() /.../module-webdriver/vendor/bin/codecept:21
PHP   3. eval() /.../module-webdriver/vendor/bin/codecept(21) : eval()'d code:6
PHP   4. {closure:/.../module-webdriver/vendor/codeception/codeception/app.php:7-46}() /.../module-webdriver/vendor/codeception/codeception/app.php:46
PHP   5. Codeception\Application->registerCustomCommands() /.../module-webdriver/vendor/codeception/codeception/app.php:38
PHP   6. Codeception\Application->readCustomCommandsFromConfig() /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Application.php:31
PHP   7. Codeception\Configuration::config($configFile = *uninitialized*) /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Application.php:63
PHP   8. file_get_contents($filename = '/.../module-webdriver/codeception.yml') /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Configuration.php:184
PHP Fatal error:  Uncaught TypeError: Codeception\Configuration::getConfFromContents(): Argument #1 ($contents) must be of type string, bool given, called in /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Configuration.php on line 185 and defined in /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Configuration.php:408
Stack trace:
#0 /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Configuration.php(185): Codeception\Configuration::getConfFromContents()
#1 /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Application.php(63): Codeception\Configuration::config()
#2 /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Application.php(31): Codeception\Application->readCustomCommandsFromConfig()
#3 /.../module-webdriver/vendor/codeception/codeception/app.php(38): Codeception\Application->registerCustomCommands()
#4 /.../module-webdriver/vendor/codeception/codeception/app.php(46): {closure}()
#5 /.../module-webdriver/vendor/bin/codecept(21) : eval()'d code(6): require('...')
#6 /.../module-webdriver/vendor/bin/codecept(21): eval()
#7 {main}
  thrown in /.../module-webdriver/vendor/codeception/codeception/src/Codeception/Configuration.php on line 408
```
  
After:
```  
PHP Warning:  file_get_contents(/.../Codeception/codeception.yml): Failed to open stream: Permission denied in /.../Codeception/src/Codeception/Configuration.php on line 197
PHP Stack trace:
PHP   1. {main}() /.../Codeception/codecept:0
PHP   2. require() /.../Codeception/codecept:7
PHP   3. {closure:/.../Codeception/app.php:7-46}() /.../Codeception/app.php:46
PHP   4. Codeception\Application->registerCustomCommands() /.../Codeception/app.php:38
PHP   5. Codeception\Application->readCustomCommandsFromConfig() /.../Codeception/src/Codeception/Application.php:31
PHP   6. Codeception\Configuration::config($configFile = *uninitialized*) /.../Codeception/src/Codeception/Application.php:63
PHP   7. file_get_contents($filename = '/.../Codeception/codeception.yml') /.../Codeception/src/Codeception/Configuration.php:197

In Configuration.php line 199:

  Failed to read /.../Codeception/codeception.yml
  
```

Reply:
Understood. So it's fine as is, thanks for clarifying.